src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -907,6 +1097,13 @@ class CreateStatementSegment(BaseSegment):
             ),
             optional=True,
         ),
+        # Next are WAREHOUSE options https://docs.snowflake.com/en/sql-reference/sql/create-warehouse.html
+        Sequence(
+            Sequence("WITH", optional=True),
+            Ref("WarehouseObjectPropertiesSegment", optional=True),
+            Ref("WarehouseObjectParamsSegment", optional=True),
+            optional=True,
```

Comment:
Since all the elements are optional, is it important to make the whole thing optional as well?

Reply:
Not really I guess. But it's consistent with the rest of the clauses in this Segment so would prefer to keep as is.